pkg/collector/collector.go

```diff
@@ -67,3 +76,26 @@ func (b *BaseCollector) Export() error {
 
 	return nil
 }
+
+// Helper function to collect output of a given kubectl command to a file. Returns kubectl's stderr output if stdout output is empty.
+func (b *BaseCollector) collectKubeResourceToFile(rootPath, fileName string, kubeCmds []string) error {
```

Comment:
I feel we can accept this as is for now and address as part of later refactor (this func is partially duplicated elsewhere anyway)

Reply:
💡 please note and remember: these points are only for author to keep in mind, in case it’s super easy or it helped them in their design to be cleaner they are welcome to take onboard but beyond.

mas long as we have captured those thinking and thoughts I am happy for any moving forward decisions, changing design is much bigger refactor then moving functions between files if the underlying workflow needs it. 

Again, thought share for Original authors to way in and decide.

Reply:
sure - just sharing my opinion too :)

Reply:
@Tatsinnit @davidkydd 
We thought it would help others if we kept the function in `pkg/collector/collector.go`  since it seemed that the rest of the repo mostly does the same thing
-- collect kubectl resource and then write to file